### PR TITLE
XW-229 don't follow redirects to empty articles

### PR DESCRIPTION
**Pull request by bognix**
Jira ticket: https://wikia-inc.atlassian.net/browse/XW-229

Related PR: https://github.com/Wikia/mercury/pull/1467

### CODE REVIEW

File: includes/wikia/api/ArticlesApiController.class.php [hunk 1]
```diff
@@ -988,7 +988,6 @@ public function getAsJson() {
 
 		if ( !empty( $articleId ) ) {
 			$article = Article::newFromID( $articleId );
-			$title = $article->getTitle();
```
Comment (rogatty) [inline includes/wikia/api/ArticlesApiController.class.php]: It was here for a reason, surely. See https://github.com/Wikia/app/pull/7796

File: includes/wikia/api/ArticlesApiController.class.php [hunk 2]
```diff
@@ -1003,9 +1002,10 @@ public function getAsJson() {
 
 		if ( $redirect !== 'no' && $article->getPage()->isRedirect() ) {
 			// false, Title object of local target or string with URL
+			/** @var Title|string|boolean $followRedirect */
 			$followRedirect = $article->getPage()->followRedirect();
 
-			if ( $followRedirect && !is_string( $followRedirect ) ) {
+			if ( $followRedirect instanceof Title && !empty($followRedirect->getArticleID()) ) {
```
Comment (rogatty) [inline includes/wikia/api/ArticlesApiController.class.php]: Missing spaces.

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 3]
```diff
@@ -338,9 +338,12 @@ public function getArticle() {
 			$article = Article::newFromID( $articleId );
 
 			if ( $title->isRedirect() ) {
-				/* @var Title $title */
-				$title = $article->getRedirectTarget();
-				$article = Article::newFromID( $title->getArticleID() );
+				/* @var Title|null $redirectTargetTitle */
+				$redirectTargetTitle = $article->getRedirectTarget();
+				if ( $redirectTargetTitle instanceof Title && !empty($redirectTargetTitle->getArticleID() ) ) {
```
Comment (rogatty) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: Missing space.

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 4]
```diff
@@ -338,10 +338,16 @@ public function getArticle() {
...
+				/* @var Title|null $redirectTargetTitle */
+				$redirectTargetTitle = $article->getRedirectTarget();
 				$data['redirected'] = true;
+				if ( $redirectTargetTitle instanceof Title && !empty($redirectTargetTitle->getArticleID() ) ) {
+					$article = Article::newFromID( $redirectTargetTitle->getArticleID() );
+					$title = $redirectTargetTitle;
+				} else {
+					$data['redirectEmptyTarget'] = true;
+				}
+
```
Comment (tomasznapieralski) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: remove extra line

File: includes/wikia/api/ArticlesApiController.class.php [hunk 5]
```diff
@@ -1003,9 +1002,10 @@ public function getAsJson() {
 
 		if ( $redirect !== 'no' && $article->getPage()->isRedirect() ) {
 			// false, Title object of local target or string with URL
+			/** @var Title|string|boolean $followRedirect */
```
Comment (tomasznapieralski) [inline includes/wikia/api/ArticlesApiController.class.php]: remove additional `*`

**Comment by vforge** [comment]
Minor styling issues, LGTM

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 6]
```diff
@@ -338,10 +338,16 @@ public function getArticle() {
 			$article = Article::newFromID( $articleId );
 
 			if ( $title->isRedirect() ) {
-				/* @var Title $title */
-				$title = $article->getRedirectTarget();
-				$article = Article::newFromID( $title->getArticleID() );
+				/* @var Title|null $redirectTargetTitle */
+				$redirectTargetTitle = $article->getRedirectTarget();
 				$data['redirected'] = true;
+				if ( $redirectTargetTitle instanceof Title && !empty($redirectTargetTitle->getArticleID() ) ) {
```
Comment (rogatty) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: Missing space.

**Comment by rogatty** [comment]
LGTM after formatting is fixed

**Pull request merged**